src/server/pbsd_main.c

```diff
@@ -2143,11 +2145,28 @@ main(int argc, char **argv)
 		if (reap_child_flag)
 			reap_child();
 #endif	/* WIN32 */
-
+		scks.active_socks = 0;
+		act_scks = 0;
+		/* count the number of active schedulers */
+		for (psched = (pbs_sched*) GET_NEXT(svr_allscheds); psched; psched = (pbs_sched*) GET_NEXT(psched->sc_link)) {
+			if (psched->scheduler_sock != -1) {
+				scks.active_socks = scks.active_socks + 1;
+			}
+		}
+		scks.socket_fd = (int *)calloc(scks.active_socks, sizeof(int));
```

Comment:
Need to validate the failure case of calloc.

Reply:
Also no need to typecast for calloc.

Reply:
done